### PR TITLE
Update search docs to rst + update hashtag docs

### PR DESCRIPTION
**Pull request by lfbrock**

### CODE REVIEW

File: source/help/getting-started/searching.rst [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+- Do not start with a number.
+- Are at least 3 characters long, not including the ``#``.
+- Are made up of alphanumeric or other unicode characters.
+- May contain dots, dashes or underscores.
+
+Examples: ``#bug``, ``#marketing``, ``#v2.1``, ``#user_testing``,
+``#per.iod``, ``#check-in``, ``#마케팅``
+
+Hashtags do not link to channels. For example, if you have a channel
```
Comment (jasonblais) [inline source/help/getting-started/searching.rst]: Wondering if we should mention that "~" can be used for channel linking and reference to that doc?

File: source/help/getting-started/searching.rst [hunk 2]
```diff
@@ -0,0 +1,115 @@
+Searching
```
Comment (jasonblais) [inline source/help/getting-started/searching.rst]: Not sure if it's just an issue in github, but this doc is rendering in raw rst format.

File: source/help/getting-started/searching.rst (hunk 1 above)
Comment (lfbrock) [inline source/help/getting-started/searching.rst]: Do you know which doc that's in? I'm having trouble finding it

File: source/help/getting-started/searching.rst (hunk 2 above)
Comment (lfbrock) [inline source/help/getting-started/searching.rst]: No idea, seemed fine on my local copy so maybe just a github issue? Not sure.

File: source/help/getting-started/searching.rst (hunk 1 above)
Comment (jasonblais) [inline source/help/getting-started/searching.rst]: Hm, tried searching for it but I couldn't find it either. Maybe we never added one

**Comment by lfbrock** [comment]
Okay I'm not sure why this doesn't render properly in github, so we should make sure to double check this doc after merging to make sure it looks correct.

File: source/help/getting-started/searching.rst [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+=========
+
+--------------
+
+The search box in Mattermost returns search results from any channel of
+which you’re a member.
+
+-  Searching multiple words returns results with any one of the words
```
Comment (esethna) [inline source/help/getting-started/searching.rst]: Default was changed to "AND"

File: source/help/getting-started/searching.rst [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+   channel archive.
+-  You can expand the search results window by clicking the “Expand”
+   button in the top right corner.
+-  Use search modifiers, like ``from:dave``, to return results only from
+   certain people or in certain channels (see below)
+
+Like many search engines, highly common words like ``the``, ``which``,
+``are`` (known as “stop words”), as well as two-letter and one-letter
+search terms, are not shown in search because they typically return too
+may results.
```
Comment (esethna) [inline source/help/getting-started/searching.rst]: Propose we link the stop words for MySQL and Postgres databases here as well as below.

```
See database documentation on `MySQL`_ and `Postgres`_ for a full list of stop words.
```

File: source/help/getting-started/searching.rst [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+   messages in Town Square that contain ``Mattermost``.
+
+“Quotation Marks”
+^^^^^^^^^^^^^^^^^
+
+Use quotation marks to return search results for exact terms.
+
+-  For example: Searching ``"Mattermost website"`` returns messages
+   containing the entire phrase ``Mattermost website`` and not messages
+   containing only ``Mattermost`` or ``website``.
```
Comment (esethna) [inline source/help/getting-started/searching.rst]: or > and

File: source/help/getting-started/searching.rst [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+alphanumeric or other unicode characters.
+
+Valid hashtags:
+
+- Do not start with a number.
+- Are at least 3 characters long, not including the ``#``.
+- Are made up of alphanumeric or other unicode characters.
+- May contain dots, dashes or underscores.
+
+Examples: ``#bug``, ``#marketing``, ``#v2.1``, ``#user_testing``,
```
Comment (esethna) [inline source/help/getting-started/searching.rst]: #v2.1 is not a valid hashtag currently. ie it does not create a link when posted

Comment (esethna) [inline source/help/getting-started/searching.rst]: I think it's because it starts with a number and contains a dot

**Comment by jasonblais** [comment]
This looks good on docs.mattermost.com

image

**Pull request merged**